Revert:

From: Roland Mc Grath <roland@hack.frob.com>
Date: Tue, 8 May 2012 17:44:57 +0000 (+0200)
Subject: Warn if user requests __FORTIFY_SOURCE but it is disabled
X-Git-Tag: glibc-2.16-tps~448
X-Git-Url: http://sourceware.org/git/?p=glibc.git;a=commitdiff_plain;h=05c2c9618f583ea4acd69b3fe5ae2a2922dd2ddc

Warn if user requests __FORTIFY_SOURCE but it is disabled

[BZ #13979]
        * include/features.h: Warn if user requests __FORTIFY_SOURCE
        checking but the checks are disabled for any reason.
---

2012-05-08  Roland Mc Grath  <roland@hack.frob.com>

        [BZ #13979]
        * include/features.h: Warn if user requests __FORTIFY_SOURCE
        checking but the checks are disabled for any reason.

--- a/include/features.h
+++ b/include/features.h
@@ -409,19 +409,15 @@
 
 #if defined _FORTIFY_SOURCE && _FORTIFY_SOURCE > 0
 # if !defined __OPTIMIZE__ || __OPTIMIZE__ <= 0
-#  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
 # elif !__GNUC_PREREQ (4, 1)
-#  warning _FORTIFY_SOURCE requires GCC 4.1 or later
 # elif _FORTIFY_SOURCE > 2 && (__glibc_clang_prereq (9, 0)		      \
 			       || __GNUC_PREREQ (12, 0))
 
 #  if _FORTIFY_SOURCE > 3
-#   warning _FORTIFY_SOURCE > 3 is treated like 3 on this platform
 #  endif
 #  define __USE_FORTIFY_LEVEL 3
 # elif _FORTIFY_SOURCE > 1
 #  if _FORTIFY_SOURCE > 2
-#   warning _FORTIFY_SOURCE > 2 is treated like 2 on this platform
 #  endif
 #  define __USE_FORTIFY_LEVEL 2
 # else
